js/experiment.js

```diff
@@ -49,60 +54,131 @@ function getPosition(elem) {
+    if (mode === 'node-mode') {
+        var position = getClickPosition(e);
+        var node = document.createElementNS(xmlns, 'rect');
+
+        console.log(position.x, position.y);
+        // check for overlaps and off the chart problems before creating
+        node.setAttributeNS(null, 'x', position.x);
+        node.setAttributeNS(null, 'y', position.y);
+        node.setAttributeNS(null, 'id', nodeId);
+        node.setAttributeNS(null, 'width', 60);
+        node.setAttributeNS(null, 'height', 30);
+        node.setAttributeNS(null, 'fill', nodeColour);
+    //    node.setAttributeNS(null, 'style', 'border: 2px solid black');
+        node.setAttributeNS(null, 'class', 'node');
+        
+        document.getElementById('mySVG').appendChild(node);
+        document.getElementById(nodeId).addEventListener('mousedown', nodeClicked, false);
+        document.getElementById(nodeId).addEventListener('mousemove', nodeMoved, false);
+        document.getElementById(nodeId).addEventListener('mouseup', nodeUnclicked, false);
+        //$("#" + nodeId).mousedown(function () {nodeClicked(nodeClicked())};);
+        nodeId = nodeId + 1;
+    }
 }
 
-function nodeSelected(elem) {
+
+function nodeClicked(e) {
+    'use-strict';
+
     if (mode  === 'erase-mode') {
-        document.getElementById("mySVG").removeChild(elem);
+        document.getElementById('mySVG').removeChild(e.currentTarget);
+    } else if (mode === 'change-mode') {
+        nodeSelected = e.currentTarget;
+        var position = getClickPosition(e);
+        nodeX = position.x;
+        nodeY = position.y;
+        console.log(nodeX, nodeY);
     }
 }
 
+
+function nodeMoved(e) {
+    'use-strict';
+
+    if (mode === 'change-mode' && nodeSelected !== null) {
+        nodeSelected = this;
+        var position = getClickPosition(e);
+        var rectX = e.currentTarget.getAttribute('x')*1;
+        var rectY = e.currentTarget.getAttribute('y')*1;
+        rectX += (position.x - nodeX);
+        rectY += (position.y - nodeY);
+        e.currentTarget.setAttribute('x', rectX);
+        e.currentTarget.setAttribute('y', rectY);
+        nodeX = position.x;
+        nodeY = position.y;
+        console.log(e.currentTarget.x.animVal.value, e.currentTarget.y.animVal.value, nodeX, nodeY);
+    }
+}
+
+
+function nodeUnclicked(e) {
+    'use-strict';
+
+    if (mode === 'change-mode') {
+        nodeSelected = null;
+        nodeX = -1;
+        nodeY = -1;
+        console.log(nodeX, nodeY);
+    }
+}
+
+
 function changeMode(id) {
-    //if (mode !== '') {
-    //  oldMode = document.getElementById(mode);
-    //  oldMode.setAttribute('border', 'border: 2px solid #008080');
-    //}
+    'use-strict';
+
+    if (mode !== '') {
+      $('#'+mode).toggleClass('clicked');
+    }
     mode = id;
-    //newMode = document.getElementById(mode);
-    //newMode.setAttribute('border', 'border: 2px solid white');
+    newMode = document.getElementById(mode);
```

Comment:
I don't see `newMode` as a global variable, nor used anywhere else in this file. What is its purpose?